main.tf

```diff
@@ -61,11 +66,12 @@ resource "aws_dynamodb_table" "main" {
   restore_date_time      = var.restore_date_time
   stream_enabled         = var.stream_enabled
   stream_view_type       = var.stream_view_type
+
   dynamic "server_side_encryption" {
```

Comment:
This shouldn't be dynamic, you should always use the aws default key and they you can choose to override it with the key created on the module or a key arn/id external to the module